File: backend/app/models/task.py
"""任务模型。"""
from datetime import datetime, timezone
from sqlalchemy import Integer, String, Text, DateTime, ForeignKey, Column, JSON
from sqlalchemy.orm import relationship
from app.core.database import Base

class Task(Base):
    __tablename__ = "tasks"
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    book_id = Column(Integer, ForeignKey("books.id", ondelete="CASCADE"), nullable=False, index=True)
    task_type = Column(String(50), default="tts", nullable=False)
    status = Column(String(50), default="pending", nullable=False, index=True)
    celery_task_id = Column(String(255), nullable=True)
    progress = Column(Integer, default=0, nullable=False)
    error_message = Column(Text, nullable=True)
    params = Column(JSON, nullable=True)
    result = Column(JSON, nullable=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), index=True)
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))
    completed_at = Column(DateTime, nullable=True)
    user = relationship("User", back_populates="tasks")
    book = relationship("Book", back_populates="tasks")

File: flutter_app/lib/models/task.dart
/// 任务模型
class Task {
  final int id;
  final int userId;
  final int bookId;
  final String taskType;
  final String status;
  final int progress;
  final String? errorMessage;
  final String? celeryTaskId;
  final String createdAt;
  final String updatedAt;
  final String? completedAt;

  Task({
    required this.id,
    required this.userId,
    required this.bookId,
    required this.taskType,
    required this.status,
    required this.progress,
    this.errorMessage,
    this.celeryTaskId,
    required this.createdAt,
    required this.updatedAt,
    this.completedAt,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json["id"] as int,
      userId: json["user_id"] as int,
      bookId: json["book_id"] as int,
      taskType: json["task_type"] as String? ?? "tts",
      status: json["status"] as String? ?? "pending",
      progress: json["progress"] as int? ?? 0,
      errorMessage: json["error_message"] as String?,
      celeryTaskId: json["celery_task_id"] as String?,
      createdAt: json["created_at"] as String? ?? "",
      updatedAt: json["updated_at"] as String? ?? "",
      completedAt: json["completed_at"] as String?,
    );
  }
}
